scripts/mempool.js

```diff
@@ -1,381 +1,391 @@
-     * @param {Number} UTXO.vout - Output position of this transaction
+     * Get the total wallet balance
      */
-    autoRemoveUTXO({ id, vout }) {
-        for (const cUTXO of this.UTXOs) {
-            // Loop given + internal UTXOs to find a match, then start the delayed removal
-            if (cUTXO.id === id && cUTXO.vout === vout) {
-                cUTXO.status = Mempool.REMOVED;
-                this.removeWithDelay(12, cUTXO);
-                return;
+    async getBalanceNew(filter) {
+        const startTime = new Date();
+        console.log('Starting calculating total balance');
+        let totBalance = 0;
+        for (let [_, tx] of this.txmap) {
+            for (let vout of tx.vout) {
+                if (this.isSpent(vout.outpoint)) {
+                    continue;
+                }
+                const UTXO_STATE = (await wallet.isMyVout(vout.script)).state;
+                if ((UTXO_STATE & filter) == 0) {
+                    continue;
+                }
+                totBalance += vout.value;
             }
         }
-        console.error(
-            'Mempool: Failed to find UTXO ' +
-                id +
-                ' (' +
-                vout +
-                ') for auto-removal!'
+        const endTime = new Date();
+        console.log(
+            'Finished calculating total balance',
+            (endTime - startTime) / 1000
         );
+        return totBalance;
     }
-
     /**
-     * Remove many UTXOs completely from our wallet, with a 12 minute delay
-     * @param {Array<UTXO>} arrUTXOs - UTXOs to remove
+     * Outpoint that we want to fetch
+     * @param {COutpoint} op
      */
-    autoRemoveUTXOs(arrUTXOs) {
-        for (const cNewUTXO of arrUTXOs) {
-            for (const cUTXO of this.UTXOs) {
-                // Loop given + internal UTXOs to find a match, then start the delayed removal
-                if (cUTXO.equalsUTXO(cNewUTXO)) {
-                    cUTXO.status = Mempool.REMOVED;
-                    this.removeWithDelay(12, cUTXO);
-                    break;
+    async hasUTXO(op, filter, onlyConfirmed) {
+        // If the outpoint is spent return false
+        if (this.isSpent(op)) {
+            return false;
+        }
+        // If we don't have the outpoint return false
+        if (!this.txmap.has(op.txid)) {
+            return false;
+        }
+        const tx = this.txmap.get(op.txid);
+        // Check if the tx is confirmed
+        if (onlyConfirmed && !tx.isConfirmed()) {
+            return false;
+        }
+        const vout = tx.vout[op.n];
+        const UTXO_STATE = (await wallet.isMyVout(vout.script)).state;
+        // Check if the UTXO has the state we wanted
+        if ((UTXO_STATE & filter) == 0) {
+            return false;
+        }
+        return true;
+    }
+    // a bit a copy and paste from getBalanceNew, TODO: remove the copy and paste
+    async getAllUTXOsWithValue(val, filter, onlyConfirmed) {
```

Comment:
```suggestion
    async getAllUTXOsWithValue({val, filter, onlyConfirmed = true|false}) {
```
Destructure and provide a default value for more readability

Reply:
this function has been removed with the last commit